sklearn/feature_selection/_rfe.py

```diff
@@ -786,11 +785,18 @@ def fit(self, X, y, groups=None):
 
         # reverse to stay consistent with before
         scores_rev = scores[:, ::-1]
-        self.cv_results_ = {}
-        self.cv_results_["mean_test_score"] = np.mean(scores_rev, axis=0)
-        self.cv_results_["std_test_score"] = np.std(scores_rev, axis=0)
-
-        for i in range(scores.shape[0]):
-            self.cv_results_[f"split{i}_test_score"] = scores_rev[i]
+        self.cv_results_ = {
+            "n_features": np.flip(
+                np.append(
+                    np.arange(
+                        start=n_features, stop=self.min_features_to_select, step=-step
+                    ),
+                    self.min_features_to_select,
+                )
+            ),
```

Comment:
Instead of re-implementing this decreasing stepping logic, I wonder if the `_rfe_single_fit` function could not be changed to return the array of `n_features` that match the score array it returns. This way we would be more confident that those two regions of the code-based cannot diverge in case of refactoring.

Or alternatively, factorize the call to:

```python
np.arange(
    start=n_features, stop=self.min_features_to_select, step=-step
)
```
in a private helper function that is used both in the `RFE` class and here.

Reply:
I made a similar comment https://github.com/scikit-learn/scikit-learn/pull/28670#pullrequestreview-1951457206, proposing to make the list of n_features directly a public attribute of `RFE`, which I think would make things even simpler that what you propose. What do you think @ogrisel ?